### PR TITLE
python-common: Add mypy testing

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

cc @jschmid1 

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

**Comment by jschmid1** [comment]
+1 for mypy usage

**Comment by sebastian-philipp** [comment]
This has an impact on make check, but its not related to QA.

File: src/python-common/tox.ini [hunk 1]
```diff
@@ -3,9 +3,11 @@ envlist = py27, py3, lint
 skip_missing_interpreters = true
 
 [testenv]
-deps=
-  pytest
-commands=py.test -v {posargs:ceph/tests}
+deps=-rrequirements.txt
+commands=python setup.py test --addopts="--mypy --mypy-ignore-missing-imports" {posargs}
```
Comment (tchaikov) [inline src/python-common/tox.ini]: does this imply python2?

Comment (sebastian-philipp) [reply src/python-common/tox.ini]: There is a special overrride for `[testenv:py27]` below

Comment (tchaikov) [reply src/python-common/tox.ini]: i still don't get it. i think it's
```
[testenv:py27]
commands=python setup.py test {posargs}
```
which overrides the global setting.

so "py3" and "lint" will be using
```
python setup.py test --addopts="--mypy --mypy-ignore-missing-imports" {posargs}
```
for testing.

Comment (sebastian-philipp) [reply src/python-common/tox.ini]: So you would prefer to have a `[testenv:py3]` here? I can do that!

Comment (sebastian-philipp) [reply src/python-common/tox.ini]: done

File: src/python-common/setup.py [hunk 2]
```diff
@@ -1,6 +1,16 @@
...
+
 from setuptools import setup, find_packages
 
 
+if sys.version_info >= (3,0):
+    mypy = ['mypy', 'pytest-mypy']
+    pytest = 'pytest >=2.1.3'
+else:
+    mypy = []
+    pytest = 'pytest >=2.1.3,<5'
```
Comment (bk201) [inline src/python-common/setup.py]: Just for a record.
https://docs.pytest.org/en/latest/py27-py34-deprecation.html

**Comment by sebastian-philipp** [comment]
Probably caused https://tracker.ceph.com/issues/42528

**Pull request merged**